Wrap color palette in fieldset with label inside of a legend

jeryj opened the pull request:
Closes #18308

Color palette popovers were not having their label read on screen readers, because they were just a div and span. This commit wrapps the toggle buttons in a fieldset with the label as a legend so it will read the label when the component receives focus.

## How has this been tested?
Manually with Chrome and Voice Over


## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [ ] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->

jeryj committed:
Wrapped color palette in fieldset with label inside of a legend

Color palette popovers were not having their label read on screen readers, because they were just a div and span. This commit wrapps the toggle buttons in a fieldset with the label as a legend so it will read the label wwhen the component receives focus.

obenland commented:
It looks like the changed mark up caused some test failures. Could you amend those to account for the new dom structure?

jeryj committed:
Updated snapshot for color-palette component

The markup for the color-palette was changed to add a fieldset and legend, so the snapshot needed to be updated for tests to pass.

jeryj commented:
Good catch, @obenland! I updated the snapshot for this component, and unit tests pass now.

jeryj committed:
Fixed color button selector paths in e2e tests.

Also updated the snapshots, as I changed the colors to be more obviously changed when watching the tests running in the browser.

tellthemachines commented on packages/e2e-tests/specs/editor/blocks/heading.test.js:
```diff
@@ -60,14 +58,15 @@ describe( 'Heading', () => {
+		await page.keyboard.type( '#7700ff' );
 		await page.click( '[data-type="core/heading"] h3' );
-		await page.waitForSelector( '.component-color-indicator[aria-label="(Color: #181717)"]' );
+		await page.waitForSelector( '.component-color-indicator[aria-label="(Color: #7700ff)"]' );
```
Any particular reason for changing the colour?

jeryj replied:
I was watching the tests via the `PUPPETEER_HEADLESS=false PUPPETEER_SLOWMO=80 npm run test-e2e:watch` command, and changing it to a color like `#181717` didn't give much of a visual indication that the color had changed. However, changing it to `#7700ff`, a bright purple, made it obvious that the color change had applied successfully :)

tellthemachines reviewed (approved):
Thanks for fixing this! Checked on VoiceOver and NVDA; the controls are now correctly being announced as a group and the legend is being read out.
